src/masternodes/rpc_vault.cpp

```diff
@@ -1911,6 +1911,72 @@ UniValue getstoredinterest(const JSONRPCRequest& request) {
     return GetRPCResultCache().Set(request, ret);
 }
 
+UniValue logstoredinterests(const JSONRPCRequest& request) {
+    RPCHelpMan{"logstoredinterests",
+               "Logs all stored interests.\n",
+               {
+
+                },
+               RPCResult{
+                       "[\"vaultId\": {\n"
+                       "  \"token\" : n,                            Token ID\n"
+                       "  \"amount\" : n,                           (amount) Token Amount\n"
+                       "  \"interestHeight\" : n,                   Height stored interest last updated\n"
+                       "  \"interestToHeight\" : n.nnnnnnnn,        Interest stored to the point of the hight value\n"
+                       "  \"interestPerBlock\" : n.nnnnnnnn,        nterest per block\n"
+                       "}] \n"
+               },
+               RPCExamples{
+                       HelpExampleCli("logstoredinterests", "") +
+                       HelpExampleRpc("logstoredinterests", "")
+               },
+    }.Check(request);
+
+    if (auto res = GetRPCResultCache().TryGet(request)) return *res;
+
+    LOCK(cs_main);
+
+    auto height = ::ChainActive().Height();
+    using VaultInfo = std::tuple<DCT_ID,CAmount,CInterestRateV3>;
+    std::map<std::string, std::vector<VaultInfo>> items;
```

Comment:
Why not add directly to UniValue and avoid iterating again after?

Reply:
Please don't do that at all, unless there is a specific reason to do so. The extra memory by the raw dataset, compared to UniValue is usually small. And so is the additional computation that they are irrelevant. Code clarity, reusability and maintenance is usually far more important to focus on. This decouples the formatting from the actual logic. 

Second, the format could something else that's not JSON at all (GRPC, for instance). Mixing it up with UniValue makes things messy.